pkg/webhook/resources/addon/validator.go

```diff
@@ -83,12 +83,26 @@ func validateUpdatedAddon(newAddon *v1beta1.Addon, oldAddon *v1beta1.Addon) erro
 		return werror.NewBadRequest("chart field cannot be changed.")
 	}
 
+	if v1beta1.AddonOperationInProgress.IsTrue(oldAddon) {
+		return werror.NewBadRequest(fmt.Sprintf("cannot perform operation, as an existing operation is in progress on addon %s", oldAddon.Name))
+	}
+
 	if newAddon.Name == vClusterAddonName && newAddon.Namespace == vClusterAddonNamespace && newAddon.Spec.Enabled {
 		return validateVClusterAddon(newAddon)
 	}
 	return nil
 }
 
+func isAddonSpecChanged(newAddon *v1beta1.Addon, oldAddon *v1beta1.Addon) bool {
+	if newAddon.Spec != oldAddon.Spec {
```

Comment:
prefer to use `reflect.DeepEqual` instead.
If not any other check, we could simplify it like this